### ISSUE TITLE
v0.1.2: Empty groups cause errors.

### ISSUE THREAD

**Issue by racknoris** [AdobeXD/xd-to-flutter-plugin#52]
On AdobeXD, the "Copy Selected" doesn't work on the new version, did work before I updated today. Tried to quit the program after updating and it didn't do anything

**Comment by gskinner** [comment]
Can you provide any reproduction steps? Is it just a specific thing that won't copy? For example, if you just draw a rectangle and copy it, does it work?

If you go to menu > plugins > development > Developer Console, are there any errors?

I just tested with a few things, and it worked for me, so I'll need more info to narrow it down.

Thanks.

**Comment by rogasciences** [comment]
Having the same issue since update, 
this is the Log

Plugin Error: Plugin Promise was rejected with: infinite loop in combineShapes
    at convertPluginErrorToString (plugins/PluginErrorUtil.js:1:198)
    at safeGetStackTrace (plugins/PluginErrorUtil.js:1:339)
    at internalFormatPluginError (plugins/PluginErrorUtil.js:1:1073)
    at internalReportPluginError (plugins/PluginErrorUtil.js:1:1171)
    at Object.reportPluginError (plugins/PluginErrorUtil.js:1:1603)
    at plugins/ScenegraphGuard.js:1:5659

**Comment by gskinner** [comment]
Can you provide any reproduction steps? Is it just a specific thing that won't copy? For example, if you just draw a rectangle and copy it, does it work?

**Comment by racknoris** [comment]
@gskinner I now created a new XD document, added an artboard, and drew a rectangle inside it. After that, I opened the developer console and saw the line:
`Loading plugin /Users/username/Library/Application Support/Adobe/Adobe XD/plugins/6eaf77ea`.
When I selected the rectangle and pressed "Copy Selected", the developer console showed: `test 1`, with no further output, and the shape did copy code to the clipboard:
```
Container(
    width: 76.0,
    height: 79.0,
    decoration: BoxDecoration(
      color: const Color(0xffbe2d2d),
      border: Border.all(width: 1.0, color: const Color(0xff707070)),
    ),
  );
```

I took the original design, selected some text and pressed "Copy selected", then I saw many lines of debug in the console, and an error at the end, so I started to delete all the shapes and pictures until I got to [this file](https://drive.google.com/file/d/1uyG6JeSwm7imHeHxxEWbaS4b0A8aqF6z/view?usp=sharing). It contains a single artboard with a text "Sample title". When you select it and "Copy Selected", you get in the console:
```
test 1
added import package:flutter_svg/flutter_svg.dart to file XDlocationicon
didn't add import ./XDlocationicon.dart because there was no current file
```

Maybe that could help? I could isolate other parts of the file and get the other errors, but it might be in the weekend

**Comment by racknoris** [comment]
This is what I get when trying to copy something in the file, prior to cleaning it:
```
test 1
added import package:flutter_svg/flutter_svg.dart to file null
Didn't add import for component 'Facebook' because the master was not found
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSsplashscreen
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSsplashscreen
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSsplashscreen
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSsplashscreen
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSsplashscreen
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSsplashscreen
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSsplashscreen
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSsplashscreen
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSsplashscreen
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSsplashscreen
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSsplashscreen
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSsplashscreen
didn't add import ./XDiPhoneXXSTutorial01.dart because there was no current file
added import package:flutter_svg/flutter_svg.dart to file null
Didn't add import for component 'Facebook' because the master was not found
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSTutorial01
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSTutorial01
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSTutorial01
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSTutorial01
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSTutorial01
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSTutorial01
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSTutorial01
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSTutorial01
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSTutorial01
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSTutorial01
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSTutorial01
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSTutorial01
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSTutorial01
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSTutorial01
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSTutorial01
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSTutorial01
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSTutorial01
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSTutorial01
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSTutorial01
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSTutorial01
added import package:adobe_xd/blend_mask.dart to file XDiPhoneXXSTutorial01
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSTutorial01
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSTutorial01
added import ./XDComponent1191.dart to file XDiPhoneXXSTutorial01
added import ./XDiPhoneXXSHome.dart to file XDiPhoneXXSTutorial01
added import package:flutter_svg/flutter_svg.dart to file null
Didn't add import for component 'Facebook' because the master was not found
added import ./XDComponent1191.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import ./XDiPhoneXXSHome.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial02
added import package:flutter_svg/flutter_svg.dart to file null
Didn't add import for component 'Facebook' because the master was not found
added import ./XDComponent1191.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import ./XDiPhoneXXSHome.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXSiPhoneXXSTutorial03
added import package:flutter_svg/flutter_svg.dart to file null
Didn't add import for component 'Elements/Map' because the master was not found
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXShome02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXShome02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXShome02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXShome02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXShome02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXShome02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXShome02
added import package:flutter_svg/flutter_svg.dart to file null
Didn't add import for component 'icons/light/close' because the master was not found
added import ./XDiPhoneXXShome03.dart to file XDiPhoneXXShome02
added import package:flutter_svg/flutter_svg.dart to file null
Didn't add import for component 'icons/light/next' because the master was not found
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXShome02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXShome02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXShome02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXShome02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXShome02
added import package:flutter_svg/flutter_svg.dart to file null
Didn't add import for component 'icons/light/menu' because the master was not found
added import ./XDMenu.dart to file XDiPhoneXXShome02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXShome02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXShome02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXShome02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXShome02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXShome02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXShome02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXShome02
added import package:flutter_svg/flutter_svg.dart to file XDiPhoneXXShome02
added import ./XDiPhoneXXSTutorial01.dart to file XDiPhoneXXShome02
added import package:flutter_svg/flutter_svg.dart to file null
Didn't add import for component 'icons/light/close' because the master was not found
added import package:flutter_svg/flutter_svg.dart to file null
added import ./XDiPhoneXXShomelistview.dart to file null
Didn't add import for component 'icons/setting' because the master was not found
added import package:flutter_svg/flutter_svg.dart to file null
added import package:flutter_svg/flutter_svg.dart to file null
Didn't add import for component 'button/fab/filters' because the master was not found
added import package:flutter_svg/flutter_svg.dart to file null
Didn't add import for component 'icons/gps' because the master was not found
Didn't add import for component 'button/fab/gps' because the master was not found
added import package:flutter_svg/flutter_svg.dart to file null
Didn't add import for component 'button/fab/gps' because the master was not found
Didn't add import for component 'button/fab/gps+filters' because the master was not found
added import ./XDiPhoneXXSscan01.dart to file null
added import package:flutter_svg/flutter_svg.dart to file null
Didn't add import for component 'camera-icon' because the master was not found
added import ./XDComponent31.dart to file XDiPhoneXXShome02
added import package:flutter_svg/flutter_svg.dart to file null
added import ./XDsearch.dart to file XDiPhoneXXShome02
added import ./XDiPhoneXXSHomesearch.dart to file XDiPhoneXXShome02
Plugin Error: Plugin Promise was rejected with: infinite loop in combineShapes
    at convertPluginErrorToString (plugins/PluginErrorUtil.js:1:198)
    at safeGetStackTrace (plugins/PluginErrorUtil.js:1:339)
    at internalFormatPluginError (plugins/PluginErrorUtil.js:1:1073)
    at internalReportPluginError (plugins/PluginErrorUtil.js:1:1171)
    at Object.reportPluginError (plugins/PluginErrorUtil.js:1:1603)
    at plugins/ScenegraphGuard.js:1:5535

```

**Comment by gskinner** [comment]
@racknoris - I really appreciate your time and input on this.

What I really need is an XD file & steps to generate this error: `Plugin Error: Plugin Promise was rejected with: infinite loop in combineShapes`

I haven't been able to reproduce it here, but it's obviously impacting people and I'd really like to fix it.

Also pinging @rogasciences in case they can provide a file.

**Comment by racknoris** [comment]
@gskinner - no problem, it's for all of us flutter developers, I'm happy to contribute.

I clean up other parts of the original file and narrowed it down to a single text element which causes the error (there may be more, but this is one of them). [Here is](https://drive.google.com/file/d/11Zp0Ai-y9d10yi67K-p2Un86BownAg-B/view?usp=sharing) a link to the file. When you click on the text "Sample" and then click "Copy Selected" in the plugin, you get this in the console:
```
test 1
Plugin Error: Plugin Promise was rejected with: infinite loop in combineShapes
    at convertPluginErrorToString (plugins/PluginErrorUtil.js:1:198)
    at safeGetStackTrace (plugins/PluginErrorUtil.js:1:339)
    at internalFormatPluginError (plugins/PluginErrorUtil.js:1:1073)
    at internalReportPluginError (plugins/PluginErrorUtil.js:1:1171)
    at Object.reportPluginError (plugins/PluginErrorUtil.js:1:1603)
    at plugins/ScenegraphGuard.js:1:5535
```

Also, I edited my previous comment, as I saw I did not copy the whole content

**Comment by gskinner** [comment]
@racknoris - thanks! That let me isolate the problem. The issue is actually with empty groups, which is something we don't have in our test files (I'll add it).

It should be an easy fix, but will require a full testing pass, so I'm hoping to submit an updated version late today, and then it will be a couple days before it's available on the plugin marketplace.

In the interim, removing any empty groups from your file should fix it.

**Comment by racknoris** [comment]
@gskinner - sounds great. Also, about the errors in my [previous](https://github.com/AdobeXD/xd-to-flutter-plugin/issues/52#issuecomment-636885110) comment before your edit- can they cause problems in copying, or the actual important message is just `Plugin error`?

**Comment by gskinner** [comment]
In this case, only the error was relevant, so I just edited it down to keep this post a bit shorter / easier to scan. :)

**Comment by racknoris** [comment]
Oh ok, now I understand
By the way, thank you for CreateJS, used it for a while few years back, and it worked great :)

**Comment by gskinner** [comment]
This should be fixed in 4c408acfd0dc6422fdd917695a23ad2031f39ef0. Will be included in v0.1.3

**Comment by gskinner** [comment]
As an aside, I still haven't figure out how to create an empty group. XD deletes groups when you remove the last item in them. It makes me feel a little better that our tests didn't catch this issue. :)

Luckily I was able to copy and paste the empty group from the repro file from @racknoris , so at least I have one in my test files now.
